src/bootstrap/builder.rs

```diff
@@ -109,6 +109,25 @@ impl RunConfig<'_> {
     }
 }
 
+/// A description of the crates in this set, suitable for passing to `builder.info`.
+///
+/// `crates` should be generated by [`RunConfig::cargo_crates_in_set`].
+pub fn crate_description(crates: Interned<Vec<String>>) -> String {
+    if crates.is_empty() {
+        return "".into();
+    }
+
+    let mut descr = String::from(": {");
+    for krate in &*crates {
+        write!(descr, "{}, ", krate.strip_prefix("-p=").unwrap()).unwrap();
+    }
+
+    descr.pop();
+    descr.pop();
+    descr.push('}');
+    descr
+}
```

Comment:
I can remove this bit from the PR if you like; it's mostly important for https://github.com/rust-lang/rust/pull/106303 which you've already approved.

Reply:
No need to do that.